glusterd2/commands/volumes/volume-option.go

```diff
@@ -113,13 +134,23 @@ func xlatorAction(c transaction.TxnCtx, txnOp txnOpType, volOp volumeOpType) err
 			reqOptions[key] = op.DefaultValue
 		}
 
-	}
-	var volinfo volume.Volinfo
-	if err := c.Get("volinfo", &volinfo); err != nil {
-		return err
+	case xlator.VolumeStart:
+		fallthrough
+
+	case xlator.VolumeStop:
+		for _, actor := range xlator.GetOptActors() {
+			if txnOp == txnDo {
+				fn = actor.Do
+			} else {
+				fn = actor.Undo
+			}
+			if err := fn(&volinfo, "", "", volOp, c.Logger()); err != nil {
```

Comment:
Concern unrelated to PR: If one of the actors fail, the undo step will run all of them. This requires undo functions to be written in certain way. The actor framework function implementations has gotten complicated over time. May be declarative daemon management will help. I don't know.